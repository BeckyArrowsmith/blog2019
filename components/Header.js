import Nav from "../components/Nav";

const Header = () => (
    <header className="Header">
        <Nav />

        <div className="container bio">
            <h1>becky.codes</h1>

            <p>Less code these days.</p>
            <p>Adventures in enabling the best of people and engineering teams.</p>
            <p><a href="http://twitter.com/beckycodes">Twitter</a> <a href="https://www.linkedin.com/in/beckyarrowsmith/">LinkedIn</a> <a href="http://github.com/beckyarrowsmith">Github</a></p>
        </div>
    </header>
);

export default Header;
