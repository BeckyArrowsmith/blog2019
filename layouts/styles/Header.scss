header {
    $bg-color: white;
    $dot-color: black;

    $dot-size: 1px;
    $dot-space: 4px;

	background:
		linear-gradient(90deg, $bg-color ($dot-space - $dot-size), transparent 1%) center,
		linear-gradient($bg-color ($dot-space - $dot-size), transparent 1%) center,
		$dot-color;
    background-size: $dot-space $dot-space;

    .container {
        * {
            background: #fff;
            padding: 0 10px;
        }
    }

    .bio {
        h1, p {
            padding-bottom: 10px;
        }

        a {
            padding: 0;
            color: #505050;
        }
    }
}