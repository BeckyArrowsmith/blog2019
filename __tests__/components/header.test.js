/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../../pages/index.js'
import Header from '../../components/Header.js'

describe('With Enzyme', () => {
    it('Header renders without crashing', () => {
      shallow(<Header />);
    });
})

describe('With Snapshot Testing', () => {
    it('Header renders without crashing', () => {
        const component = renderer.create(<Header />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})