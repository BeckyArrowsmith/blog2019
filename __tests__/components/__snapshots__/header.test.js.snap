// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Header renders without crashing 1`] = `
<header>
  <a
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    style={
      Object {
        "marginRight": 15,
      }
    }
  >
    Home
  </a>
  <a
    href="/about"
    onClick={[Function]}
    onMouseEnter={[Function]}
    style={
      Object {
        "marginRight": 15,
      }
    }
  >
    About
  </a>
</header>
`;
